TigerConnectDemoApp/app/src/main/java/com/tigertext/ttandroid/sample/conversation/ConversationFragment.java

```diff
@@ -0,0 +1,380 @@
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        // If the adapter is not null and the item count is not 0, then we know messages have been loaded.
+        // mark any messages as read in case the Activity is going from onResume due to the phone being in sleep state
+        // or app being backgrounded.
+        if (conversationAdapter != null && conversationAdapter.getItemCount() != 0) {
+            // This is how you mark a conversation as read
+            TT.getInstance().getConversationManager().markConversationAsRead(mRosterEntry);
+        }
+    }
+
+    @Override
+    public void setUserVisibleHint(boolean isVisibleToUser) {
+        super.setUserVisibleHint(isVisibleToUser);
+        if (isVisibleToUser) {
+            // This is how you mark a conversation as read
+            TT.getInstance().getConversationManager().markConversationAsRead(mRosterEntry);
+        }
+    }
+
+    private void setupRecyclerView() {
+        conversationAdapter = new ConversationAdapter(this);
+        recyclerConversation.setAdapter(conversationAdapter);
+        linearLayoutManager = new LinearLayoutManager(getContext());
+        linearLayoutManager.setStackFromEnd(true);
+        recyclerConversation.setLayoutManager(linearLayoutManager);
+    }
+
+    private void setupViewModel() {
+        ConversationViewModel conversationViewModel = ViewModelProviders.of(getActivity()).get(ConversationViewModel.class);
+        conversationViewModel.init();
+        mRosterEntry = conversationViewModel.getSelectedRosterEntry().getValue();
+        conversationViewModel.getSelectedRosterEntry().observe(this, rosterEntry -> {
+            if (rosterEntry == null) return;
+
+            mRosterEntry = rosterEntry;
+            conversationName.setText(rosterEntry.getDisplayName());
+        });
+
+        conversationViewModel.updateMessages(mRosterEntry, PAGE_SIZE, null);
+        conversationViewModel.getMessages().observe(this, messages -> {
+            if (messages == null) return;
+
+            Timber.d("Messages size: %s", messages.size());
+            conversationAdapter.updateMessages(messages);
+        });
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        TTPubSub.getInstance().removeListeners(this, listeners);
+    }
+
+
+    /**
+     *  This is how you auto forward your messages to this user
+     */
+    private void autoForwardToThisUser() {
+        Timber.d("autoForwardToThisUser");
+        Map<SettingType, Object> modifiedSettingsMap = new ArrayMap<>(1);
+        modifiedSettingsMap.put(SettingType.DND_AUTO_FORWARD_RECEIVER, mRosterEntry.getId());
+        // Call this method to notify our Organization Manager to modify the organization settings
+        // and set this user as the recipient of our auto forwarded messages
+        TT.getInstance().getOrganizationManager().modifyOrganizationSettings(mRosterEntry.getOrgId(), modifiedSettingsMap, null);
```

Comment:
It _might_ be a good idea to put a listener here so that this fully documents the usage, how to know if it was successful/failure